blocks/datasets/streams.py

```diff
@@ -314,6 +314,40 @@ def _cache(self):
             cache.extend(data)
 
 
+class SortMapping(object):
+    """Callable class for creating sort mappings without nesting functions."""
+    def __init__(self, key, reverse):
+        self.key = key
+        self.reverse = reverse
+
+    def __call__(self, x):
+        indices = [i for (v, i) in
+                   sorted(((v, i) for (i, v) in enumerate(x[0])),
```

Comment:
Slightly confused here. If you do `sorted(((v, i) for (i, v) in enumerate(x[0])), key=self.key)`, the key is applied to `(v, i)`? I would expect it to be applied to `v` directly. So if I want to sort a batch by length, I would expect to pass `len` as key, but in this case I would need to pass something like `lambda x: len(x[0])`, right?